### PR TITLE
Fix FloatType equaling ConstantFloatType

### PR DESCRIPTION
**Pull request by jlherren**
Also add some equality tests for types.

### CODE REVIEW

File: tests/PHPStan/Type/BooleanTypeTest.php [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+	public function testAccepts(): void
+	{
+		$booleanType = new BooleanType();
+
+		$this->assertTrue($booleanType->accepts(new BooleanType(), true)->yes());
+		$this->assertTrue($booleanType->accepts(new ConstantBooleanType(true), true)->yes());
+		$this->assertTrue($booleanType->accepts(new NullType(), true)->no());
+		$this->assertTrue($booleanType->accepts(new MixedType(), true)->yes());
+		$this->assertTrue($booleanType->accepts(new FloatType(), true)->no());
+		$this->assertTrue($booleanType->accepts(new StringType(), true)->no());
```
Comment (ondrejmirtes) [inline tests/PHPStan/Type/BooleanTypeTest.php:21]: should be done the same way as other tests - e.g. dataAccepts + testAccepts

**Review by ondrejmirtes — CHANGES_REQUESTED** [review]
Otherwise great 👍 :)

File: tests/PHPStan/Type/BooleanTypeTest.php (hunk 1 above)
Comment (jlherren) [reply tests/PHPStan/Type/BooleanTypeTest.php:21]: Ah... right. I copied it from `IntegerTypeTest`, that's why it's like that. I can fix them both.

**Comment by ondrejmirtes** [comment]
Awesome, thank you!

**Pull request merged**